setuptools/build_meta.py

```diff
@@ -171,7 +171,7 @@ def run_setup(self, setup_script='setup.py'):
         with _open_setup_script(__file__) as f:
             code = f.read().replace(r'\r\n', r'\n')
 
-        exec(compile(code, __file__, 'exec'), locals())
+        exec(code, locals())
```

Comment:
The only difference I can think here is that if the setup script has a syntax error, compile would fail so no instruction gets executed. I don't know how `exec` works, but it could be the case it gets to execute something before failing... I don't think this is relevant though because the PEP 517 hooks are supposed to be executed in isolated subcommands, so it is fine for the hook to crash.

However, I might be failing to see other problems...

Alternatively we could also try.

```python
        try:
            exec(code, locals())
        except Exception:
            exec(compile(code, __file__, 'exec'), locals())
```